Fix bug 871, 'IcMetadata.Info can be stale'

mrelac opened the pull request:
Android platform only: When a new track is selected that has a Metadata header with a metadataInterval of -1 (i.e. C.LENGTH_UNSET), onMetadata() is never called; thus, the stale metadata info from the last track is not cleared. This fix clears the metadata info under this condition.

ryanheise committed:
Merge branch 'feature/uuid_4' into minor

ryanheise committed:
Merge branch 'feature/uuid_4' into major

ryanheise committed:
Merge branch 'feature/uuid_4' into feature/treadmill

goviral-ma committed:
update

goviral-ma committed:
update

ryanheise committed:
Add setAllowsExternalPlayback on iOS/macOS.

ryanheise committed:
Merge branch 'feature/ios_external_playback' into interface

ryanheise committed:
Android extractor options, proxyless headers/user-agent.

ryanheise committed:
Proxyless headers/user-agent on iOS/macOS.

ryanheise committed:
Java compile error.

ryanheise committed:
CHANGELOG

Chaphasilor committed:
fix conversion between milliBel and deciBel

- previously, the set or reported gain would be in Bel instead of deciBel (10 times as great)
- this change fixes the conversion so that e.g. if the user provides the value in deciBel, any (Android) audio effect will correctly reflect the gain
- this closes #1089

Commit:
Add test to check that supplied request headers aren't overwritten by proxy handler

Commit:
Ensure user supplied headers are written on top of default headers

Commit:
Bump version and desc for #1100 in change log

MinSeungHyun committed:
Fix memory leak by cancelling subscriptions on dispose

MinSeungHyun committed:
Fix memory leak on android

ryanheise committed:
Fix NSNull vs nil check.

ryanheise committed:
Mention useProxyForRequestHeaders in the README.

ryanheise committed:
Implement index-based seeking on Android.

ryanheise committed:
Fix Android NPE in getDuration()

ryanheise committed:
Update platform interface CHANGELOG

ryanheise committed:
Implement preferPreciseDurationAndTiming on iOS/macOS.

ryanheise committed:
Pass through missing methods in just_audio_background.

ryanheise committed:
Merge branch 'feature/extractor_options' into minor

ryanheise committed:
Merge branch 'feature/extractor_options' into major

ryanheise committed:
Merge branch 'feature/extractor_options' into feature/treadmill

ryanheise committed:
Merge branch 'fix-unable-to-overwrite-request-headers-reproducer' of https://github.com/ctedgar/just_audio into ctedgar-fix-unable-to-overwrite-request-headers-reproducer

ryanheise committed:
Formatting

ryanheise committed:
Merge branch 'ctedgar-fix-unable-to-overwrite-request-headers-reproducer' into minor

ryanheise committed:
Merge branch 'ctedgar-fix-unable-to-overwrite-request-headers-reproducer' into major

ryanheise committed:
Merge branch 'ctedgar-fix-unable-to-overwrite-request-headers-reproducer' into feature/treadmill

idy committed:
fix b/1121

ryanheise committed:
Polish README.

ryanheise committed:
Merge branch 'minor' of https://github.com/idy/just_audio into fix/web_play_queue

ryanheise committed:
Clean up, version bump, CHANGELOG.

ryanheise committed:
Merge branch 'fix/web_play_queue' into minor

ryanheise committed:
Merge branch 'fix/web_play_queue' into major

ryanheise committed:
Merge branch 'fix/web_play_queue' into feature/treadmill

ryanheise committed:
Migrate to package:web

ryanheise committed:
Merge branch 'fix-millibel-decibel-conversion' of https://github.com/Chaphasilor/just_audio into Chaphasilor-fix-millibel-decibel-conversion

ryanheise committed:
Merge branch 'Chaphasilor-fix-millibel-decibel-conversion' into major

ryanheise committed:
Android setup instructions for SDK 34

ryanheise committed:
Merge branch 'fix/sdk34docs' into minor

ryanheise committed:
Merge branch 'fix/sdk34docs' into major

ryanheise committed:
Merge branch 'fix/sdk34docs' into feature/treadmill

ryanheise committed:
Implement useLazyPreparation on iOS/macOS

ryanheise committed:
Fix seek to index UI update on iOS/macOS.

ryanheise committed:
Add sequence index to stream errors on iOS/macOS.

ryanheise committed:
Add sequence index to stream errors on Android.

ryanheise committed:
Add details to PlayerException. Update README/meta files.

ryanheise committed:
Merge branch 'feature/treadmill' into minor

ryanheise committed:
Merge branch 'feature/treadmill' into major

mathisfouques committed:
Changes on just_audio and the background package. Adding tags on convenience functions, updating doc, and putting an assert with correct debug message before the cast.

mathisfouques committed:
Fix small typo error

jonmarkhall committed:
Fix: Exception when calling insert() or removeAt() on ConcatenatingAudioSource due to shuffleOrder not being updated

canxin121 committed:
[Fix] preferPreciseDurationAndTiming config unused in file Uri

danielwinkler committed:
Update just_audio_web.dart

danielwinkler committed:
Move set cross origin to platform

danielwinkler committed:
update library versions and dependencies

danielwinkler committed:
Expose web cross origin as setWebCrossOrigin

ryanheise committed:
Merge branch 'minor' into feature/web_package

ryanheise committed:
Merge branch 'feature/web_package' into minor

ryanheise committed:
Merge branch 'feature/web_package' into major

danielwinkler committed:
Merge branch 'minor' into feature/web-cross-origin

danielwinkler committed:
update versions after merging from minor

danielwinkler committed:
pubspec quote formatting

danielwinkler committed:
Set just_audio version to 0.9.38

danielwinkler committed:
Add cookie information to the readme

ryanheise committed:
Bump web to <0.6.0

ryanheise committed:
Merge branch 'feature/web-cross-origin' of https://github.com/danielwinkler/just_audio into feature/bump_web

ryanheise committed:
Polishing crossorigin API and docs.

ryanheise committed:
Gradle and multidex changes.

ryanheise committed:
Bump just_audio_background example to Android SDK 34

ryanheise committed:
Merge branch 'feature/bump_web' into major

ryanheise committed:
Merge branch 'feature/bump_web' into minor

ryanheise committed:
Add setWebCrossOrigin to just_audio_background.

ryanheise committed:
Merge branch 'minor' of https://github.com/canxin121/just_audio into canxin121-minor

ryanheise committed:
Update changelog and version.

ryanheise committed:
Merge branch 'canxin121-minor' into major

ryanheise committed:
Merge branch 'canxin121-minor' into minor

ryanheise committed:
Merge branch 'minor' of https://github.com/mathisfouques/just_audio into feature/tag_convenience

ryanheise committed:
Polish comments/assertions.

ryanheise committed:
Merge branch 'minor' of https://github.com/goviral-ma/just_audio into feature/tag_convenience

ryanheise committed:
Add comma.

ryanheise committed:
Update changelog.

ryanheise committed:
Merge branch 'feature/tag_convenience' into major

ryanheise committed:
Merge branch 'feature/tag_convenience' into minor

Gyugyu94 committed:
Updated the comments related to useLazyPreparation in ConcatenatingAudioSource

fedelubax commented:
@ryanheise any plans to merge this?

ryanheise commented:
I had plans to change the API altogether to be less tied to ICY metadata, however I think it's reasonable to probably merge this now. But have you tested it yourself and found it to work well?

ryanheise committed:
Support rxdart 0.28.x

ryanheise committed:
Merge branch 'fix/rxdart_bump' into minor

ryanheise committed:
Merge branch 'fix/rxdart_bump' into major

ryanheise committed:
Support rxdart 0.28.x in background.

ryanheise committed:
Merge branch 'fix/bg_rxdart_bump' into minor

ryanheise committed:
Merge branch 'fix/bg_rxdart_bump' into major

ryanheise committed:
Fix JDK 21 compile error.

ryanheise committed:
Merge branch 'fix/jdk_21_compilation' into major

ryanheise committed:
Merge branch 'fix/jdk_21_compilation' into minor

ali2236 committed:
bump package:web version constraint

ali2236 committed:
bump just_audio_web & just_audio versions

ali2236 committed:
change web constraint to >=0.5.1 <2.0.0

ali2236 committed:
Add changes to CHANGELOG.md

ryanheise committed:
Fix versioning and warnings.

ryanheise committed:
Merge branch 'ali2236-minor' into major

ryanheise committed:
Merge branch 'ali2236-minor' into minor

ryanheise committed:
Fix main CHANGELOG

ryanheise committed:
Merge branch 'ali2236-minor' into major

ryanheise committed:
Merge branch 'ali2236-minor' into minor

Commit:
Migrate to ExoPlayer to Media3

Commit:
Update version number

Commit:
Update changelog

Commit:
Correct changelog entry, add GitHub username

arolleaguekeng committed:
Update README.md

The named parameter 'second' isn't defined. 
correct parameter is seconds

SleepySquash committed:
Bootstrap

ryanheise committed:
Shorten CHANGELOG

ryanheise committed:
Merge branch 'SleepySquash-1326-improve-web-methods-signature' into major

ryanheise committed:
Merge branch 'SleepySquash-1326-improve-web-methods-signature' into minor

ryanheise committed:
Merge branch 'migrate_to_media3' of https://github.com/CornCobsApps/just_audio into CornCobsApps-migrate_to_media3

ryanheise committed:
Fix stop() to cause play() to return on iOS.

ryanheise committed:
Merge branch 'fix/ios_stopped_play' into major

ryanheise committed:
Merge branch 'fix/ios_stopped_play' into minor

cristian1980 committed:
don't broadcast players states when unloading the plugin

cristian1980 committed:
fix compile error

MinSeungHyun committed:
Merge branch 'ryanheise:minor' into fix/memory-leak

cristian1980 committed:
Fix macos compile error

ryanheise committed:
Code formatting and CHANGELOG

ryanheise committed:
Merge branch 'cristian1980-minor' into major

ryanheise committed:
Merge branch 'cristian1980-minor' into minor

MinSeungHyun committed:
Merge branch 'ryanheise:minor' into fix/memory-leak

MinSeungHyun committed:
Specify types for StreamSubscription

ryanheise committed:
Merge branch 'patch-1' of https://github.com/arolleaguekeng/just_audio into arolleaguekeng-patch-1

ryanheise committed:
Align comments

ryanheise committed:
Merge branch 'arolleaguekeng-patch-1' into minor

ryanheise committed:
Merge branch 'arolleaguekeng-patch-1' into major

ryanheise committed:
Merge branch 'minor' of https://github.com/Gyugyu94/just_audio into Gyugyu94-minor

ryanheise committed:
Wrap comments

ryanheise committed:
Merge branch 'Gyugyu94-minor' into major

ryanheise committed:
Merge branch 'Gyugyu94-minor' into minor

MinSeungHyun committed:
Merge branch 'minor' into fix/memory-leak

ryanheise committed:
Update CHANGELOG

ryanheise committed:
Merge branch 'silgam-fix/memory-leak' into major

ryanheise committed:
Merge branch 'silgam-fix/memory-leak' into minor

ryanheise committed:
Bump gradle to 8.5.0

ryanheise committed:
Merge branch 'feature/bump_gradle' into major

ryanheise committed:
Merge branch 'feature/bump_gradle' into minor

dganzella committed:
feat: Add set sink id Method

dganzella committed:
fix: fix references to use local development

dganzella committed:
fix: fixing yaml

dganzella committed:
fix: fix refs

dganzella committed:
fix: use empty string to reset to default

dganzella committed:
fix: make sure operation is asynchronous

dganzella committed:
fix: fix future

ryanheise committed:
Fix NPE in load on iOS/macOS

ryanheise committed:
Improve exception reporting on Android.

ryanheise committed:
Merge branch 'fix/ios_load_npe' into minor

ryanheise committed:
Merge branch 'fix/ios_load_npe' into major

ryanheise committed:
Merge branch 'fix-cat-insertremove' of https://github.com/jonmarkhall/just_audio into jonmarkhall-fix-cat-insertremove

ryanheise committed:
Fix shuffleOrder in just_audio_background for ConcatenatingMoveRequest.

ryanheise committed:
Merge branch 'jonmarkhall-fix-cat-insertremove' into minor

ryanheise committed:
Merge branch 'jonmarkhall-fix-cat-insertremove' into major

ryanheise committed:
Migrate to media3 ExoPlayer 1.4.1 on Android.

dganzella committed:
feat: add setWebSinkId method

dganzella committed:
Merge branch 'ryanheise:minor' into setWebSinkId

dganzella committed:
fix: fix spacing

dganzella committed:
fix: fix spacing

dganzella committed:
fix: reverting spacing

dganzella committed:
fix: having trouble with yaml

dganzella committed:
fix: fix variable names

ryanheise committed:
Merge branch 'feature/exoplayer_media3' into CornCobsApps-migrate_to_media3

ryanheise committed:
Merge branch 'CornCobsApps-migrate_to_media3' into major

ryanheise committed:
Merge branch 'CornCobsApps-migrate_to_media3' into minor

ryanheise committed:
Use sharedDarwinSource

ryanheise committed:
Micrate to SwiftPM

ryanheise committed:
Add missing import to LoadControl.h. Update example Podfile.

ryanheise committed:
Bump version, CHANGELOG, README.

ryanheise committed:
Merge branch 'feature/swiftpm' into minor

ryanheise committed:
Merge branch 'feature/swiftpm' into major

ryanheise committed:
Optimise tests.

ryanheise committed:
Implement missing IdleAudioPlayer methods. Add webCrossOrigin getter.

ryanheise committed:
Test setAllowsExternalPlayback/setWebCrossOrigin/setGain/asyncMessages

ryanheise committed:
Merge branch 'feature/coverage' into minor

ryanheise committed:
Merge branch 'feature/coverage' into major

ryanheise committed:
Merge branch 'setWebSinkId' of https://github.com/PrecisionReporters/just_audio into PrecisionReporters-setWebSinkId

ryanheise committed:
Futureproof sinkId, add missing implementations, tests, bump versions, CHANGELOGs.

ryanheise committed:
Merge branch 'PrecisionReporters-setWebSinkId' into minor

ryanheise committed:
Merge branch 'PrecisionReporters-setWebSinkId' into major

ryanheise committed:
Fix SwiftPM support on macOS.

ryanheise committed:
Merge branch 'fix/swiftpm_macos' into minor

ryanheise committed:
Merge branch 'fix/swiftpm_macos' into major

ryanheise committed:
New playlist and error APIs.

ryanheise committed:
Add iOS/bg implementations for new playlist/error API

ryanheise committed:
Fix bugs in error handling.

ryanheise committed:
Add errorStream.

ryanheise committed:
Convert state diagram to use new playlist API.

ryanheise committed:
Bump min flutter version to 3.27.0, AGP to 8.5.2.

ryanheise committed:
Sync sdk with flutter version.

ryanheise committed:
Add .cxx to .gitignore

ryanheise committed:
Bump version to 4.5.0 for just_audio_platform_interface.

ryanheise committed:
Update example to use just_audio_media_kit

ryanheise committed:
Bump audio_session dep to 0.2.0.

ryanheise committed:
Update background to audio_session 0.2.0

hpoul committed:
Allow audio_session 0.2.0

audio_session 0.2.x does not introduce any breaking changes to the API, but only to the iOS setup. Hence no code changes are required.

ryanheise committed:
Merge branch 'patch-1' of github.com:hpoul/just_audio into hpoul-patch-1

ryanheise committed:
Merge branch 'hpoul-patch-1' into feature/playlist

ryanheise committed:
Update example to use latest just_audio_media_kit.

ryanheise committed:
Relax audio_session constraint on just_audio_background.

ryanheise committed:
Fix error when running under flutter test.

ryanheise committed:
Merge branch 'feature/playlist' into major

ryanheise committed:
Merge branch 'major' into minor

ryanheise committed:
Fix unhandled PlayerInterruptedException.

ryanheise committed:
Merge branch 'fix/unhandled_exception' into minor

ryanheise committed:
Synchronize playlist API calls. Improve load interruption logic.

ryanheise committed:
Improve position accuracy over Bluetooth on Android.

ryanheise committed:
Downgrade synchronized to support older flutter/dart SDKs.

ryanheise committed:
Merge branch 'fix/async_load' into minor

ryanheise committed:
Fix rxdart errors on lower bound version.

ryanheise committed:
Fix playing/playbackEvent emission order issue.

ryanheise committed:
Add stop button to example.

ryanheise committed:
Fix MissingPluginException in dispose on iOS.

ryanheise committed:
Merge branch 'fix/event_order' into minor

ryanheise committed:
Add PlayerEvent.

ryanheise committed:
Merge branch 'fix/player_event' into minor

rserro committed:
add clearPlayback to dismiss the device player

rserro committed:
rm unecessary code

rserro committed:
idle before broadcasting

ryanheise committed:
Fix play interrupted by load on web.

ryanheise committed:
Merge branch 'fix/web_play_load' into minor

ryanheise committed:
Merge branch 'patch-1' of github.com:rserro/just_audio into rserro-patch-1

ryanheise committed:
Update version/changelog

ryanheise committed:
Merge branch 'rserro-patch-1' into minor

ryanheise committed:
Fix pending timers bug in unit tests.

ryanheise committed:
Fix NPE in Android position broadcast.

ryanheise committed:
Mention AGP 8.6/8.7 incompatibility in README.

ryanheise committed:
Merge branch 'fix/pending_timers' into minor

ryanheise committed:
Fix bug on simultaneous loads.

ryanheise committed:
Fix simultaneous load bug on iOS/macOS.

Abestanis committed:
Fix the default index and position when loading a source

These got accidentally changed in 1b210605341b55064a1ced3c5f91396bf0b8b3f9.

ryanheise committed:
Fix dispose crash on iOS.

ryanheise committed:
Merge branch 'fix/load_defaults' of github.com:Abestanis/just_audio into Abestanis-fix/load_defaults

ryanheise committed:
Use convenience getter for initial seek values.

ryanheise committed:
Add test for initial seek values, update CHANGELOG.

ryanheise committed:
Merge branch 'Abestanis-fix/load_defaults' into fix/simul_loads

ryanheise committed:
Merge branch 'fix/simul_loads' into minor

ryanheise committed:
Add androidAudioOfflinePreferences parameter to AudioPlayer()

ryanheise committed:
Disable Android audio offload by default. Add docs.

ryanheise committed:
Merge branch 'feature/offload' into minor

mrelac committed:
Undesired behaviour: On Android, at app startup, initial seek and play to playlist index > 0 always plays url at index 0.

mrelac committed:
Merge branch 'minor' of https://github.com/mrelac/just_audio_icy into minor

mrelac committed:
Added print statement for currentIndex.

mrelac committed:
Updated print statements.
